packages/path_provider/path_provider_ios/ios/Classes/FLTPathProviderPlugin.m

```diff
@@ -3,47 +3,41 @@
 // found in the LICENSE file.
 
 #import "FLTPathProviderPlugin.h"
+#import "messages.g.h"
 
-NSString *GetDirectoryOfType(NSSearchPathDirectory dir) {
+static NSString *GetDirectoryOfType(NSSearchPathDirectory dir) {
   NSArray *paths = NSSearchPathForDirectoriesInDomains(dir, NSUserDomainMask, YES);
   return paths.firstObject;
 }
 
+@interface FLTPathProviderPlugin () <FLTPathProviderApi>
+@end
+
 @implementation FLTPathProviderPlugin
 
 + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
-  FlutterMethodChannel *channel =
-      [FlutterMethodChannel methodChannelWithName:@"plugins.flutter.io/path_provider_ios"
-                                  binaryMessenger:registrar.messenger];
-  [channel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
-    if ([@"getTemporaryDirectory" isEqualToString:call.method]) {
-      result([self getTemporaryDirectory]);
-    } else if ([@"getApplicationDocumentsDirectory" isEqualToString:call.method]) {
-      result([self getApplicationDocumentsDirectory]);
-    } else if ([@"getApplicationSupportDirectory" isEqualToString:call.method]) {
-      result([self getApplicationSupportDirectory]);
-    } else if ([@"getLibraryDirectory" isEqualToString:call.method]) {
-      result([self getLibraryDirectory]);
-    } else {
-      result(FlutterMethodNotImplemented);
-    }
-  }];
-}
-
-+ (NSString *)getTemporaryDirectory {
-  return GetDirectoryOfType(NSCachesDirectory);
+  FLTPathProviderPlugin *plugin = [[FLTPathProviderPlugin alloc] init];
+  FLTPathProviderApiSetup(registrar.messenger, plugin);
 }
 
-+ (NSString *)getApplicationDocumentsDirectory {
+- (nullable NSString *)getApplicationDocumentsPathWithError:
+    (FlutterError *_Nullable __autoreleasing *_Nonnull)error {
   return GetDirectoryOfType(NSDocumentDirectory);
 }
```

Comment:
I understand if you don't want to mess with this in this PR, but if this is switched to the `NSFileManager` API then we actually could pass back a useful error:

```
[NSFileManager.defaultManager URLForDirectory:NSCachesDirectory inDomain:NSUserDomainMask appropriateForURL:nil create:YES error:&error];
```
`create:YES` because I see the comment:
```
      // Ensure the directory exists before returning it, for consistency with
      // other platforms.
```